pages/api/dpu/setup.ts

```diff
@@ -1,33 +1,40 @@
 import { NextApiRequest, NextApiResponse } from 'next';
 import { removeRepoConfigForInstallIdForOwner } from '../../../utils/db/repos';
-import { SetupReposArgs, removePreviousInstallations, saveSetupReposInDb } from '../../../utils/db/setupRepos';
+import { removePreviousInstallations, saveSetupReposInDb, SetupReposArgs } from '../../../utils/db/setupRepos';
 import { getUserIdByTopicName } from '../../../utils/db/users';
 import { publishMessage } from '../../../utils/pubsub/pubsubClient';
+import rudderStackEvents from '../events';
 
 const setupHandler = async (req: NextApiRequest, res: NextApiResponse) => {
 	console.info("[setupHandler]Saving setup info in db...");
 	const jsonBody = req.body;
-	if (!Array.isArray(jsonBody.info)) {
-		console.error("[setupHandler] Invalid request body, 'info' is missing or not an array");
-		res.status(400).json({ "error": "Invalid request body" });
-		return;
-	}
 
 	// get user_id for the given install_id
 	const userId = await getUserIdByTopicName(jsonBody.installationId).catch((error: any) => {
 		console.error("[setupHandler/getUserIdByTopicName] Failed to fetch userId from the database.", error);
 	});
 	if (!userId) {
 		console.error(`[setupHandler/getUserIdByTopicName] NO userId found for topic name: ${jsonBody.installationId} from database.`);
+		const eventProperties = { ...jsonBody.info, response_status: 404 };
 		res.status(404).json({ "error": "No userId found for given installationId" });
+		rudderStackEvents.track("", "", 'setup', { type: 'setup-repos', eventStatusFlag: 0, eventProperties });
+		return;
```

Comment:
I think it is the right decision to fail the API call if this fails. But we should keep in mind that we want to keep an eye on this -- this will lead to more failure in the setup API calls than the current number.